src/dotnet/Chat.UI.Blazor/Services/OnboardingUI.cs

```diff
@@ -53,7 +53,7 @@ public async Task<bool> TryShow()
         // We give it 5 seconds to complete, otherwise it won't be shown
         using var cts = _lastTryShowCts = new CancellationTokenSource(TimeSpan.FromSeconds(5));
         try {
-            shouldBeShown = await ShouldBeShown(cts.Token);
+            shouldBeShown = await ShouldBeShown(cts.Token).ConfigureAwait(false);
```

Comment:
Should not we use `ConfigureAwait(true)` here to invoke `ModalUI.Show` later in the the original context?

Reply:
done